### ISSUE TITLE
undefined method `keys' for nil:NilClass in request specs

### ISSUE THREAD

**Issue by cpjmcquillan** [splitwise/super_diff#161]
When integrating Capybara helpers with integration (request) specs, 
*and* using `super_diff`, this gem seems to blow up/hang if there is
a failing test.

The typical behaviour I've seen in real repos is the failure hanging for
a long time, but in this example repo it blows up straight away.

For some reason, a hash passed to [`insert_hash_inspection_of`](https://github.com/mcmire/super_diff/blob/master/lib/super_diff/object_inspection/inspection_tree.rb#L74) is `nil`,
and a `NoMethodError` is raised.

I've created a [minimal example repo to demonstrate the issue](https://github.com/cpjmcquillan/super_diff_capybara_request_specs), 
with https://github.com/cpjmcquillan/super_diff_capybara_request_specs/commit/ab000d495cbb759bd4ee8594f5577b1246163061 being the
offending commit.

I'm happy to help issue a fix, but I'm not familiar with the gem so don't
really know where to begin.

Any help would be much appreciated!

**Comment by mcmire** [comment]
Hey @cpjmcquillan sorry for not replying to this sooner. Thank you for the report. I will take a look at this when I get a chance!

**Comment by cpjmcquillan** [comment]
Hey @mcmire, sorry I should have added a comment when closing!

While I only ran into this error when using `super_diff` (and this might indeed be a bug worth solving), I think the root cause lay elsewhere.

Once I resolved a misconfiguration (i.e the matchers were included correctly), this was no longer an issue.

For reference: https://github.com/thoughtbot/action_dispatch-testing-integration-capybara/issues/3#issuecomment-1325451473
